titus_isolate/utils.py

```diff
@@ -1,8 +1,26 @@
 import logging
+import time
+from threading import Thread
+
+import schedule
 
 from titus_isolate import log
 
+SCHEDULING_SLEEP_INTERVAL = 1
+
 
 def get_logger(level=logging.INFO):
     log.setLevel(level)
     return log
+
+
+def start_periodic_scheduling():
+    worker_thread = Thread(target=__schedule_loop)
+    worker_thread.daemon = True
+    worker_thread.start()
+
+
+def __schedule_loop():
+    while True:
+        schedule.run_pending()
+        time.sleep(SCHEDULING_SLEEP_INTERVAL)
```

Comment:
there is an (undocumented) `Scheduler.idle_seconds` (see [here](https://schedule.readthedocs.io/en/stable/_modules/schedule.html#Scheduler)). Maybe it should be favored over a 1 second polling?

Reply:
That method is informational.  It tells you how long until the next scheduled execution.  It does not execute it.  `run_pending` must be called in order to actually execute scheduled tasks.  I will reduce the sleep interval to `0.1` seconds to reduce the amount of possible skew between scheduled and actual execution

Reply:
oh.... I get it.  Sleep that long.

Reply:
I meant something like:
```python
while True:
    schedule.run_pending()
    time.sleep(schedule.default_scheduler.idle_seconds())
```
to avoid extraneous context switches? There's a tradeoff in reducing `SCHEDULING_SLEEP_INTERVAL`: it might improve the precision of `TitusPmc.duration` at the expense of more useless context switches to `titus-isolate`

Reply:
Fixed.

Reply:
:+1: